clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/chunk/ChunkAssembler.java

```diff
@@ -24,129 +20,156 @@
-        return new ValueBytesAndSchemaId(
-            null,
-            mapperValue.schemaId,
-            mapperValue.replicationMetadataVersionId,
-            mapperValue.replicationMetadataPayload);
+      if (reusedMapperValue.valueType.equals(MapperValueType.DELETE)) {
+        if (latestChunkedValueManifest != null) {
+          // Ignore older entries since a more recent manifest is discovered.
+          continue;
+        }
+        /**
+         * The latest event is a delete event.
+         */
+        if (reusedMapperValue.replicationMetadataPayload.remaining() != 0) {
+          return new ValueBytesAndSchemaId(
+              null,
+              reusedMapperValue.schemaId,
+              reusedMapperValue.replicationMetadataVersionId,
+              reusedMapperValue.replicationMetadataPayload);
+        }
+        return null;
       }
-      return null;
-    } else if (mapperValue.schemaId > 0) {
-      return new ValueBytesAndSchemaId(
-          ByteUtils.extractByteArray(mapperValue.value),
-          mapperValue.schemaId,
-          mapperValue.replicationMetadataVersionId,
-          mapperValue.replicationMetadataPayload);
-    } else if (mapperValue.schemaId == AvroProtocolDefinition.CHUNKED_VALUE_MANIFEST.getCurrentProtocolVersion()) {
-      final ChunkedValueManifest chunkedValueManifest = manifestSerializer.deserialize(
-          ByteUtils.extractByteArray(mapperValue.value),
-          AvroProtocolDefinition.CHUNKED_VALUE_MANIFEST.getCurrentProtocolVersion());
-
-      return assembleLargeValue(keyBytes, chunkedValueManifest, chunkBytes, mapperValue);
-    } else {
-      throw new VeniceException(
-          String.format(
-              "Unhandled case with chunked key suffix %s and value schema ID %d",
-              deserializeChunkedKeySuffix(mapperValue.chunkedKeySuffix),
-              mapperValue.schemaId));
-    }
-  }
-
-  private ValueBytesAndSchemaId assembleLargeValue(
-      byte[] keyBytes,
-      final ChunkedValueManifest manifest,
-      List<byte[]> allChunkBytes,
-      KafkaInputMapperValue manifestMapperValue) {
-    byte[][] valueChunks = new byte[manifest.keysWithChunkIdSuffix.size()][];
-    ByteBuffer[] chunkKeySuffixes = new ByteBuffer[manifest.keysWithChunkIdSuffix.size()];
-    int startPosition, suffixLength;
-    ByteBuffer byteBuffer;
-    for (int i = 0; i < manifest.keysWithChunkIdSuffix.size(); i++) {
-      byteBuffer = manifest.keysWithChunkIdSuffix.get(i);
-      startPosition = byteBuffer.position() + keyBytes.length;
-      suffixLength = byteBuffer.remaining() - keyBytes.length;
-      chunkKeySuffixes[i] = ByteBuffer.wrap(byteBuffer.array(), startPosition, suffixLength);
-    }
 
-    int totalByteCount = 0;
-    KafkaInputMapperValue chunk = null;
-    int chunksFound = 0;
-    for (byte[] chunkBytes: allChunkBytes) {
-      if (chunksFound == valueChunks.length) {
-        break;
+      if (reusedMapperValue.schemaId > 0) {
```

Comment:
nit: maybe we could add more java doc in the future to indicate this means it is a non-chunked put value.

Reply:
Actually, there are some javadoc below.